components/Common/Contents/FolderList/FolderList.tsx

```diff
@@ -1,90 +1,89 @@
+const FolderList = ({
+    onOverviewFolderButtonClick,
+    onFilteredFolderButtonClick,
 }: FolderCollectionProps) => {
-    const [currentFolder, setCurrentFolder] = useState("전체");
+    // Think currentFolderId와 currentFolderName을 객체로 관리하는 편이 옳은가?
+    // 객체로 관리하다면 확장성이 높아질까? const [currentFolder, setCurrentFolder] = useState({id: 0, name: "전체"}); 이런 느낌?
+
+    const router = useRouter();
+    const { folderId } = router.query as UrlQuery;
+    const { folderList } = useGetFolderList();
+    const { userId } = useGetUserId();
+
+    const [currentFolderId, setCurrentFolderId] = useState(0);
+    const [currentFolderName, setCurrentFolderName] = useState<string>("전체");
+
+    // URL 생성
     const sharingUrl = useRef("");
 
     const createSharingUrl = (userId: number, folderId: number) => {
         sharingUrl.current = `${window.location.origin}/shared?user=${userId}&folder=${folderId}`;
     };
 
-    const setCurrentFolderToOverview = () => {
-        setCurrentFolder("전체");
-        onOverviewCardButtonClick();
-    };
-
-    const setCurrentFolderToSelected = (
-        folderName: string,
-        folderId: number
-    ) => {
-        setCurrentFolder(folderName);
-        onFilteredCardButtonClick(folderId);
-        if (userData !== undefined) {
-            createSharingUrl(userData.id, folderId);
-        }
-    };
-
-    if (folderData === undefined || userData === undefined) {
-        return null;
-    }
+    const { handleOverviewFolderClick, handleFolderClick } =
+        useHandleFolderClick(
+            folderList,
+            setCurrentFolderName,
+            setCurrentFolderId,
+            onOverviewFolderButtonClick,
+            onFilteredFolderButtonClick,
+            createSharingUrl,
+            folderId,
+            userId
+        );
 
     return (
         <>
             <FolderWrapper>
                 <div>
-                    <FolderButtonWrapper>
-                        <FolderButtonBox
+                    <FolderButtonContainer>
+                        <FolderButton
                             type="button"
-                            onClick={setCurrentFolderToOverview}
+                            $isClicked={0 === currentFolderId ? true : false}
+                            onClick={handleOverviewFolderClick}
                         >
                             전체
-                        </FolderButtonBox>
-                        {folderData.map((folder) => {
+                        </FolderButton>
+                        {folderList?.map((folder: FolderInterface) => {
                             return (
-                                <FolderButtonBox
+                                <FolderButton
                                     type="button"
+                                    $isClicked={
+                                        folder.id === currentFolderId
+                                            ? true
+                                            : false
```

Comment:
`===` 비교는 boolean을 반환하니 삼항조건연산자 대신에 간단히 아래와 같이 해도 동일합니다
```suggestion
folder.id === currentFolderId
```